Packet++/src/TcpReassembly.cpp

```diff
@@ -23,13 +23,44 @@
 namespace pcpp
 {
 
-	static timeval timespecToTimeval(const timespec& in)
+	static timeval timePointToTimeval(const std::chrono::time_point<std::chrono::high_resolution_clock>& in)
 	{
-		timeval out;
-		TIMESPEC_TO_TIMEVAL(&out, &in);
+		auto duration = in.time_since_epoch();
+
+		auto seconds = std::chrono::duration_cast<std::chrono::seconds>(duration);
+		auto nanoseconds = std::chrono::duration_cast<std::chrono::nanoseconds>(duration) % 1000000000;
+
+		struct timeval out;
+		out.tv_sec = seconds.count();
+		out.tv_usec = nanoseconds.count() / 1000;
```

Comment:
Why do we calculate the nanosecond precision here if we are just going to convert them to microseconds? As far as I see, the nanosecond precision duration object is not used anywhere else in this function.

Reply:
Good catch! Fixed in https://github.com/seladb/PcapPlusPlus/pull/1591/commits/91f6b390a2c23412c301f6183ba634491106feec